File: MvvmIbatis/src/main/java/com/zhongjh/mvvmibatis/http/cookie/CookieJarImpl.java
package com.zhongjh.mvvmibatis.http.cookie;

import com.zhongjh.mvvmibatis.http.cookie.store.CookieStore;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import okhttp3.Cookie;
import okhttp3.CookieJar;
import okhttp3.HttpUrl;

/**
 * 调用Cookie方法
 *
 * @author zhongjh
 * @date 2022/3/30
 */
public class CookieJarImpl implements CookieJar {

    private final CookieStore cookieStore;

    public CookieJarImpl(CookieStore cookieStore) {
        if (cookieStore == null) {
            throw new IllegalArgumentException("cookieStore can not be null!");
        }
        this.cookieStore = cookieStore;
    }

    @Override
    public synchronized void saveFromResponse(@NotNull HttpUrl url, @NotNull List<Cookie> cookies) {
        cookieStore.saveCookie(url, cookies);
    }

    @NotNull
    @Override
    public synchronized List<Cookie> loadForRequest(@NotNull HttpUrl url) {
        return cookieStore.loadCookie(url);
    }

    public CookieStore getCookieStore() {
        return cookieStore;
    }
}

File: MvvmIbatis/src/main/java/com/zhongjh/mvvmibatis/http/cookie/CookieJarImpl.kt
package com.zhongjh.mvvmibatis.http.cookie

import com.zhongjh.mvvmibatis.http.cookie.store.CookieStore
import okhttp3.Cookie
import okhttp3.CookieJar
import okhttp3.HttpUrl

/**
 * 调用Cookie方法
 *
 * @author zhongjh
 * @date 2022/3/30
 */
class CookieJarImpl(cookieStore: CookieStore?) : CookieJar {
    private val cookieStore: CookieStore
    @Synchronized
    override fun saveFromResponse(url: HttpUrl, cookies: List<Cookie>) {
        cookieStore.saveCookie(url, cookies)
    }

    @Synchronized
    override fun loadForRequest(url: HttpUrl): List<Cookie> {
        return cookieStore.loadCookie(url)
    }

    init {
        requireNotNull(cookieStore) { "cookieStore can not be null!" }
        this.cookieStore = cookieStore
    }
}